syftlib/src/main/java/org/openmined/syft/execution/SyftJob.kt

```diff
@@ -99,59 +109,39 @@ class SyftJob(
         jobStatusProcessor.offer(JobStatusMessage.JobCycleRejected(responseData.timeout))
     }
 
-    fun downloadData(workerId: String) {
-        if (trainingParamsStatus.get() != DownloadStatus.NOT_STARTED) {
-            Log.d(TAG, "download already running")
+    fun downloadData(
+        workerId: String,
+        responseData: CycleResponseData.CycleAccept
+    ) {
+        if (cycleStatus.get() != CycleStatus.ACCEPTED) {
+            throwError(IllegalStateException("Cycle not accepted. Download cannot start"))
             return
         }
-        Log.d(TAG, "beginning download")
-        trainingParamsStatus.set(DownloadStatus.RUNNING)
-
-        requestKey?.let { request ->
-
-            networkDisposable.add(
-                Single.zip(
-                    getDownloadables(
-                        workerId,
-                        request
-                    )
-                ) { successMessages ->
-                    successMessages.joinToString(
-                        ",",
-                        prefix = "files ",
-                        postfix = " downloaded successfully"
-                    )
-                }
-                        .compose(config.networkingSchedulers.applySingleSchedulers())
-                        .subscribe(
-                            { successMsg: String ->
-                                Log.d(TAG, successMsg)
-                                trainingParamsStatus.set(DownloadStatus.COMPLETE)
-                                jobStatusProcessor.offer(
-                                    JobStatusMessage.JobReady(
-                                        model,
-                                        plans,
-                                        clientConfig
-                                    )
-                                )
-                            },
-                            { e -> jobStatusProcessor.onError(e) }
-                        )
+        if (jobDownloader.status == DownloadStatus.NOT_STARTED) {
+            jobDownloader.downloadData(
+                workerId,
+                config,
+                responseData.requestKey,
+                networkDisposable,
+                jobStatusProcessor,
+                responseData.clientConfig,
+                plans,
+                model,
+                protocols
             )
-        } ?: throw IllegalStateException("request Key has not been set")
+        }
     }
 
     /**
      * report the results back to PyGrid
      */
-    fun report(diff: State) {
-        val requestKey = requestKey
```

Comment:
This would be public API call. The user does not have any reference to request key. We would need to store request key somewhere for all future actions.